kafka/protocol/types.py

```diff
@@ -182,7 +182,7 @@ def encode(self, items):
         if items is None:
             return Int32.encode(-1)
         return b''.join(
-            [Int32.encode(len(items))] +
+            [Int32.encode(len(list(items)))] +
```

Comment:
I think we can be a little more efficient here to avoid looping over items 2x:
```
encoded_items = [self.array_of.encode(item) for item in items]
return b''.join(
    [Int32.encode(len(encoded_items))] +
    encoded_items
)
```